cores/esp8266/FS.cpp

```diff
@@ -121,6 +121,22 @@ const char* File::name() const {
     return _p->name();
 }
 
+String File::readString()
+{
+    String ret;
+    ret.reserve(size() - position());
+    char temp[256];
+    int countRead = readBytes(temp, sizeof(temp));
+    while (countRead > 0)
```

Comment:
You're forgetting to read the last chunk when it has less than sizeof(temp). Eg.: consider a file with size 257bytes, this code will only read the first 256, and the last one will be dropped.
You have to calculate the delta remaining.

Reply:
I fixed it, thank you @devyte